app/src/main/java/org/stacks/app/ui/auth/signup/ChooseUsernameViewModel.kt

```diff
@@ -37,8 +37,14 @@ class ChooseUsernameViewModel
     private val errors = BroadcastChannel<Error>(1)
 
     init {
+
+        usernameSubmitted
+            .filter { it.isEmpty() || it.contains(" ") }
```

Comment:
Could be a `String.isValidUsername()` to avoid repeating the rules, and missing one somewhere.